services/src/main/java/org/keycloak/services/x509/NginxProxySslClientCertificateLookupFactory.java

```diff
@@ -13,12 +15,30 @@
 
 public class NginxProxySslClientCertificateLookupFactory extends AbstractClientCertificateFromHttpHeadersLookupFactory {
 
+    private final static Logger logger = Logger.getLogger(NginxProxySslClientCertificateLookupFactory.class);
+
     private final static String PROVIDER = "nginx";
 
+    protected final static String TRUST_PROXY_VERIFICATION = "trust-proxy-verification";
+
+    protected boolean trustProxyVerification = false;
+
+    @Override
+    public void init(Config.Scope config) {
+        super.init(config);
+        trustProxyVerification = config.getBoolean(TRUST_PROXY_VERIFICATION, false);
+        logger.tracev("{0}: ''{1}''", TRUST_PROXY_VERIFICATION, trustProxyVerification);
+    }
+
     @Override
     public X509ClientCertificateLookup create(KeycloakSession session) {
-        return new NginxProxySslClientCertificateLookup(sslClientCertHttpHeader,
-                sslChainHttpHeaderPrefix, certificateChainLength, session);
+        if (trustProxyVerification) {
```

Comment:
Humm ... Looks like the logic is inverted? We should return NginxProxyTrustedClientCertificateLookup if the options is set to `true`, right?

Reply:
@pedroigor you saw nothing :p, no it's not :p